src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala

```diff
@@ -132,17 +132,12 @@ abstract class ClassfileParser {
     finally loaders.parentsLevel -= 1
   }
 
-  def parse(file: AbstractFile, root: Symbol): Unit = {
-    debuglog("[class] >> " + root.fullName)
-
+  def parse(file: AbstractFile, clazz: Symbol, module: Symbol): Unit = {
```

Comment:
This allows callers to pass unrelated symbols. Maybe some warning in a comment is in order (I assume calls to `companionModule` & friends are also "forbidden" inside this method. If so, also worth a comment.